### PR TITLE
Updated UGDG - More user centric and include proposed save feature for tags

### LINKED ISSUE #191: Resolve DG comments

### LINKED ISSUE #239: Rewrite sections to be more user centric
3.3, 3.5.5.1, 3.5.5.2 and 5.1 need rewriting to be more user-focused.

### PR DESCRIPTION
**Pull request by yAOwzers**
Updated UG - Section 3.3., 3.5.5.1., 3.5.5.2.
Included DG - Section 4.7 [Proposed] save feature for tags

### CODE REVIEW

**Comment by yAOwzers** [comment]
To update the UML diagrams for 4.2.2. markAsDone and 3.6. Storage

File: docs/UserGuide.md [hunk 1]
```diff
@@ -209,12 +210,14 @@ You are now in timetable mode
 
 ### 3.3. Personalised Messages  
 
-A personalised message will be printed out to the user whenever a particular command has been entered by the user.
-Personalised messages will include a random generated encouraging message followed by the name provided by the user upon launch of Zer0Note.
+We have included personalised messages to compliment your works!
+
+A personalised message will be printed out to you in the command line interface whenever you enter a particular command.
+Personalised messages will include a random generated encouraging message followed by the name provided by you upon launch of Zer0Note.
```
Comment (franceneee) [inline docs/UserGuide.md]: Nice! But I think "Personalised messages will include a random generated encouraging message followed by the name provided by you upon launching Zer0Note."/"Personalised messages will include a random generated encouraging message followed by the name provided by you upon the launch of Zer0Note." is more accurate?

**Comment by yAOwzers** [comment]
Updated diagrams in DG, 3.6. and 4.2.2.

**Comment by yAOwzers** [comment]
Claiming code in process

File: docs/UserGuide.md [hunk 2]
```diff
@@ -833,16 +801,18 @@ N/CS2101/Chapter 1:$ list
...
 that do not belong to a selected notebook can be found. There are two ways to do so:
 
 #### 3.5.5.1. Finding by keyword
 
-Finds all notebooks, sections and pages that contain the keyword in their title.  The search is not case-sensitive.
+If there is a particular keyword in the title of the various notebooks, sections and pages that you are looking for, 
+this find by keyword feature will certainly help!
+
+This feature finds all notebooks, sections and pages that contain the keyword in their title. 
+The search is not case-sensitive.
```
Comment (franceneee) [inline docs/UserGuide.md:805]: nice!

File: docs/UserGuide.md (hunk 1 above)
Comment (NeilBaner) [reply docs/UserGuide.md]: Agreed.

**Review by franceneee — APPROVED** [review]

**Pull request merged**